onboard/nucleo_bridge/I2C.cpp

```diff
@@ -0,0 +1,43 @@
+#include "I2C.h"
+
+void I2C::init()
+{
+    file = open("/dev/i2c-1", O_RDWR);
+    if (file == -1)
+    {
+        printf("failed to open i2c bus\n");
+        exit(1);
+    }
+}
+
+void I2C::transact(uint8_t addr, uint8_t cmd, uint8_t writeNum, uint8_t readNum, uint8_t *writeBuf, uint8_t *readBuf)

```

Comment:
Please make sure you have class and function level comments throughout all files. I'm stopping flagging them at this point.

Reply:
Implemented